test/Conversion/PipelineToHW/test_outlined.mlir

```diff
@@ -359,11 +352,43 @@ hw.module @testControlUsage(%arg0: i32, %go : i1, %clk: i1, %rst: i1) -> (out0:
 // CHECK:           hw.output %[[VAL_5]], %[[VAL_6]] : i32, i1
 // CHECK:         }
 hw.module @testWithStall(%arg0: i32, %go: i1, %stall : i1, %clk: i1, %rst: i1) -> (out0: i32, out1: i1) {
-  %0:2 = pipeline.scheduled(%arg0) stall %stall clock %clk reset %rst go %go : (i32) -> (i32) {
-  ^bb0(%arg0_0: i32, %s0_valid : i1):
-    pipeline.stage ^bb1 regs(%arg0_0 : i32)
+  %0:2 = pipeline.scheduled(%a0 : i32 = %arg0) stall(%s = %stall) clock(%c = %clk) reset(%r = %rst) go(%g = %go) -> (out: i32) {
+    pipeline.stage ^bb1 regs(%a0 : i32)
   ^bb1(%1: i32, %s1_valid : i1):  // pred: ^bb1
     pipeline.return %1 : i32
   }
   hw.output %0#0, %0#1 : i32, i1
 }
+
+// -----
+
+// CHECK-LABEL:  hw.module @testNaming_MyPipeline(%in0: i1, %enable: i1, %clk: i1, %rst: i1) -> (out0: i1, valid: i1) {
+// CHECK:    %testNaming_MyPipeline_s0.out0, %testNaming_MyPipeline_s0.valid = hw.instance "testNaming_MyPipeline_s0" @testNaming_MyPipeline_s0(in0: %in0: i1, enable: %enable: i1, clk: %clk: i1, rst: %rst: i1) -> (out0: i1, valid: i1)
+// CHECK:    hw.output %testNaming_MyPipeline_s0.out0, %testNaming_MyPipeline_s0.valid : i1, i1
+// CHECK:  }
+
+// CHECK-LABEL:  hw.module @testNaming_MyPipeline_s0(%in0: i1, %enable: i1, %clk: i1, %rst: i1) -> (out0: i1, valid: i1) {
+// CHECK:    %0 = comb.add %in0, %in0 {sv.namehint = "myAdd"} : i1
+// CHECK:    %s0_myAdd_reg = seq.compreg.ce sym @s0_myAdd_reg %0, %clk, %enable : i1
```

Comment:
Stupid question.  Is there no way to add a name hint to a register itself, rather than changing the name of the register in the IR?

Reply:
As in is there a way to add a namehint to a stage-separating register at the Pipeline level, which would propagate through the HW/Seq lowering? (which also would override any `sv.namehint`-related inferred name from a register source value).
i.e.
```mlir
  %0:2 = pipeline.scheduled "MyPipeline"(%a0 : i1 = %go) clock(%c = %clk) reset(%r = %rst) go(%g = %go) -> (out: i1) {
    // The register will be named after the namehint on the source operation.
    %add = comb.add %a0, %a0 {sv.namehint = "myAdd"} : i1
    pipeline.stage ^bb1 regs("MyReg" : %add : i1)
```

would emit as

```mlir
%MyReg = seq.compreg.ce sym @MyReg %0, %clk, %enable : i1
```